components/ipfs/ipfs_service.cc

```diff
@@ -159,15 +159,28 @@ IpfsService::~IpfsService() {
   Shutdown();
 }
 
+// static
+void IpfsService::MigrateProfilePrefs(PrefService* pref_service) {
+  if (pref_service->HasPrefPath(kIPFSAutoRedirectToConfiguredGateway)) {
+    return;
+  }
+  pref_service->SetBoolean(
+      kIPFSAutoRedirectToConfiguredGateway,
+      pref_service->GetBoolean(kIPFSAutoRedirectDNSLink) ||
+          pref_service->GetBoolean(kIPFSAutoRedirectGateway));
```

Comment:
i think this should prompt if only one of them was on, see comment here: https://github.com/brave/security/issues/1315#issuecomment-1595239760